### PR TITLE
Disabled extended ACLs for the `build` directory.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

We've been having build issues depending on the configuration of the host we've been building on. Since [`rpmbuild` sets `umask` to 022 by default in the `%prep` section](http://ftp.rpm.org/max-rpm/ch-rpm-b-command.html#S2-RPM-B-COMMAND-BP-OPTION), it seems that the host's extended ACLs are the source of issue, which manifested in the following way:
```
(...)
WARN[0067] Installing:                                  
WARN[0067] systemd-rpm-macros noarch       250.3-17.cm2     local-repo     6.01k    22.85k 
WARN[0067]                                              
WARN[0067] Total installed size:   6.01k                
WARN[0067] Total download size:  22.85k                 
WARN[0067] Testing transaction                          
WARN[0067] Found 1 problems                             
WARN[0067] file /usr/lib/rpm/macros.d from install of systemd-rpm-macros-250.3-17.cm2.noarch conflicts with file from package rpm-build-4.18.0-3.cm2.x86_64  
...
```

This is an output from the build of `mdadm`, which `BuildRequires: systemd-rpm-macros` and `systemd-rpm-macros` was built earlier on the host machine, while the `rpm-build` package was pre-populated from PMC. Comparing the PMC and built versions of `systemd-rpm-macros` for `/usr/lib/rpm/macros.d`:

```
rpm -qp --dump built/systemd-rpm-macros-250.3-17.cm2.noarch.rpm
(...)
/usr/lib/rpm/macros.d 0 1690576370 0000000000000000000000000000000000000000000000000000000000000000 040775 root root 0 0 0 X
(...)

rpm -qp --dump pmc/systemd-rpm-macros-250.3-16.cm2.noarch.rpm
(...)
/usr/lib/rpm/macros.d 0 1690528863 0000000000000000000000000000000000000000000000000000000000000000 040755 root root 0 0 0 X
...
```

Notice the difference between `040775` and `040755`.

###### Change Log  

- Removed extended ACLs for the `build` directory.

###### Does this affect the toolchain?  

Yes.

###### Test Methodology

- [Full pipeline build.](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=399568&view=results)

### CODE REVIEW

File: toolkit/scripts/srpm_pack.mk [hunk 1]
```diff
@@ -39,6 +39,10 @@ $(call create_folder,$(BUILD_DIR))
 $(call create_folder,$(BUILD_SRPMS_DIR))
 $(call create_folder,$(SRPM_BUILD_CHROOT_DIR))
 
+# Host's extended ACLs influence the default permissions of the
+# files inside the built RPMs. Disabling them for the build directory.
+$(call shell_real_build_only, setfacl -bn $(BUILD_DIR))
```
Comment (anphel31) [inline toolkit/scripts/srpm_pack.mk:44]: do you think any of the other directories would need the same fix?

**Review by anphel31 — APPROVED** [review]

File: toolkit/scripts/srpm_pack.mk (hunk 1 above)
Comment (PawelWMS) [reply toolkit/scripts/srpm_pack.mk:44]: The chroots where we build the packages are inside `$(BUILD_DIR)`, so this should be enough and a [PR check pipeline test](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=399535&view=results) has finally succeeded with this tweak. If we find more issues later, we can always update this change.

**Review by dmcilvaney — APPROVED** [review]

File: toolkit/scripts/srpm_pack.mk (hunk 1 above)
Comment (dmcilvaney) [reply toolkit/scripts/srpm_pack.mk:44]: Do we want this in toolkit.mk or utils.mk? It's not really unique to srpm_packing, and we will run the `$(call shell_real_build_only,...` functions before we build regardless of which `*.mk` file they are in (they are run at parse time, not execution time).

Comment (PawelWMS) [reply toolkit/scripts/srpm_pack.mk]: I added the call next to where `BUILD_DIR` is created to minimize any build issues, but I don't like `sprm_pack.mk` as a place for that as well. I'm not sure why it even creates `BUILD_DIR`. I agree that `tools.mk` is better.

**Pull request merged**